File: docs-examples/example-java/src/test/java/io/micronaut/mqtt/docs/serdes/ProductClient.java
package io.micronaut.mqtt.docs.serdes;

// tag::imports[]
import io.micronaut.mqtt.annotation.Topic;
import io.micronaut.mqtt.v5.annotation.MqttPublisher;
import io.micronaut.messaging.annotation.MessageBody;
// end::imports[]

import io.micronaut.context.annotation.Requires;

@Requires(property = "spec.name", value = "ProductInfoSerDesSpec")
// tag::clazz[]
@MqttPublisher
public interface ProductClient {

    @Topic("product")
    void send(@MessageBody ProductInfo data);
}
// end::clazz[]

File: test-suite-kotlin/src/test/kotlin/io/micronaut/mqtt/docs/serdes/ProductClient.kt
package io.micronaut.mqtt.docs.serdes

// tag::imports[]
import io.micronaut.mqtt.v5.annotation.MqttPublisher
import io.micronaut.messaging.annotation.MessageBody
import io.micronaut.mqtt.annotation.Topic
// end::imports[]

import io.micronaut.context.annotation.Requires

@Requires(property = "spec.name", value = "ProductInfoSerDesSpec")
// tag::clazz[]
@MqttPublisher
interface ProductClient {

    @Topic("product")
    fun send(@MessageBody data: ProductInfo)
}
// end::clazz[]
